%YAML: 1.0
event:
  topic: "/dvs/events"
  freq: 30 # hz
  fx: 294.57392232
  fy: 293.41504063
  cx: 172.98762549
  cy: 120.20746418
  resolution: [346, 260]
  deltaT: 0.01 #1e-1 #6e-2 #5e-3 #5e-2 #5e-3 #5e-3 #5e-2 #1.5e-2 #5e-2 #5e-2 #5e-4 #2e-2 #5e-5
  t1: 0.005 #8e-2 #3e-2 #2.5e-3 #2.5e-2 #2.5e-3 #2.5e-2 #1e-2 #4e-2 #5.5e-2 #5.5e-2 #3e-4 #2.5e-2 #3e-5
  mesh_size: 4 #7 #7 #2 #11 # value must be an even number. actual radius is [mesh_size - 1]
  timeshift_cam_imu: 0.010493761582439365
  show_events: true
  k1: -0.48691636
  k2: -0.26707268
  p1: 0.020893022
  p2: -0.00201776
  k3: -0.08930555
  t1_count: 10000 #12500 # 15000 #10000 # 2000
  t2_count: 5000 # 3000
  
# 6-13 modify before
# t1_count: 6000 #12500 # 15000 #10000 # 2000
# t2_count: 3000 # 3000
# ratio_inliers: 0.15

# 应用低通滤波器
smooth: 0.1
filter_num: 0
median_radius: 0
ignore_polarity: false #true #false
use_gauss: false
# ratio_inliers: 0.25
ratio_inliers: 0.35 # for mesh size = 4

radar_config: "/home/hao/Desktop/radar-event-new/src/TwistEstimator/config/radar_param_dji.yaml"

radar:
  topic: "/radar/data"
  type: "ars548" # ars548
  Extrinsics:
    time_offset: 0.0
    Trans: [-0.0, 0.0, 0.0]
    Rot: [1, 0, 0,
          0, 1, 0,
          0, 0, 1]

radar_event:
  T_re:              # Radar to event camera external parameters
    - [0.0,  -1.0,  0.0,  0.04077303]
    - [0.0,   0.0,  -1.0,  -0.02330925]
    - [1.0,  0.0,  0.0,  -0.16198989]
    - [0.0,  0.0,  0.0,  1.0]

event_extrincs:
  topic: "/dvs/events"
  Extrinsics:
    time_offset: 0.0
    Trans: [0.04077303, -0.02330925, -0.16198989]
    Rot: [0.0286651, -0.9992392, -0.02644507,
          -0.04739285, 0.0277848, -0.99848982,
          0.99846494, -0.0273685, 0.04815325]

# [0.0286651,  -0.04739285,  0.99846494,
# 0.9992392,   0.0277848,  -0.0273685,d
# -0.02644507,  0.99848982,  0.04815325]


# Estimator
knot_distance: 0.2 #0.2 #0.2 #0.06 # 0.2 #0.06 #0.3 #0.15 # 0.3 #0.1 #0.055 #0.03     #[s]
update_every_k_knot: 4 #4  #[knot]
linear_weight: 200.0 # 20.0 # 8 #1.2 # best 1.2 #100 # 1.4 #1.0 # best 1.0 #1.0
linear_w_weight: 0.85 # 2 # best 1 # 400
linear_R_weight: 0
omega_weight: 400.0 # 160 # 40.0 #0.06 # 3 # best 0.06 #1000 # 0.06 #0.05 #0.08 #0.15 #0.05 # best 0.05
omega_w_weight: 0.0001 #1e-6 #15 # best 10 x 100 #12 #10 #10 # 10 # best 10 # 5

## Tight couple best (no cauchy loss, options.min_lm_diagonal = 1.0, no optimize for linear velocity and bias in EventFlowFactor)
# knot_distance: 0.2 #0.2 #0.2 #0.06 # 0.2 #0.06 #0.3 #0.15 # 0.3 #0.1 #0.055 #0.03     #[s]
# update_every_k_knot: 4 #4  #[knot]
# linear_weight: 200.0 # 20.0 # 8 #1.2 # best 1.2 #100 # 1.4 #1.0 # best 1.0 #1.0
# linear_w_weight: 0.85 # 2 # best 1 # 400
# linear_R_weight: 0
# omega_weight: 400.0 # 160 # 40.0 #0.06 # 3 # best 0.06 #1000 # 0.06 #0.05 #0.08 #0.15 #0.05 # best 0.05
# omega_w_weight: 0.0001 #1e-6 #15 # best 10 x 100 #12 #10 #10 # 10 # best 10 # 5

accmulate_delta_T: 0.01

# linear_weight: 10
# linear_w_weight: 1
# linear_R_weight: 0
# omega_weight: 1
# omega_w_weight: 1
## Tight couple initial best
# linear_weight: 10
# linear_w_weight: 1
# linear_R_weight: 0
# omega_weight: 1
# omega_w_weight: 1
## for enlarge residual of angular velocity
# linear_weight: 2.0
# linear_w_weight: 1
# linear_R_weight: 0
# omega_weight: 20.0
# omega_w_weight: 1

# velocity estimation
# linear_weight: 1.2
# linear_w_weight: 10
# linear_R_weight: 0
# omega_weight: 5
# omega_w_weight: 10

# # pose estimation in SE3
# linear_weight: D1.2
# linear_w_weight: 1
# linear_R_weight: 0
# omega_weight: 0.06 # 1
# omega_w_weight: 12 # 15

# # pose estimation in R6
# linear_weight: 2 # 5
# linear_w_weight: 0.5 # 2
# linear_R_weight: 0
# omega_weight: 1.0 # 0.1 # 0.8 # 0.04
# omega_w_weight: 0.3 #1e20 #1e24 # 15 1.5e20

# pose loose estimation in R6
# linear_weight: 1 # 5
# linear_w_weight: 1 # 2
# linear_R_weight: 0
# omega_weight: 25.0 # 15 #10.0 # 0.1 # 0.8 # 0.04
# omega_w_weight: 0.8 #1e20 #1e24 # 15 1.5e20

# JUST for Linear
# linear_weight: 2 # 5
# linear_w_weight: 0.5 # 2
# linear_R_weight: 0
# omega_weight: 0.5 # 0.1 # 0.8 # 0.04
# omega_w_weight: 0.5 #1e20 #1e24 # 15 1.5e20

use_fej: true
use_prior: true

# 设置分组优化
use_order_opti: false  # use under Ceres Version 2.x

# only for angular back-end
# linear_weight: 3
# linear_w_weight: 1
# linear_R_weight: 0
# omega_weight: 0.5 # 0.1 # 0.8 # 0.04
# omega_w_weight: 1e60 #1e20 #1e24 # 15 1.5e20

## dji best
# linear_weight: 1.2
# linear_w_weight: 1
# omega_weight: 0.06
# omega_w_weight: 12 / 10

## dji best2
# linear_weight: 1.2
# linear_w_weight: 1
# omega_weight: 1
# omega_w_weight: 15

## dji best2-2
# linear_weight: 8
# linear_w_weight: 2
# linear_R_weight: 0
# omega_weight: 0.06
# omega_w_weight: 60

## dji last
# linear_weight: 8
# linear_w_weight: 2
# linear_R_weight: 0
# omega_weight: 0.1
# omega_w_weight: 65

## dji last
# linear_weight: 8
# linear_w_weight: 2
# linear_R_weight: 0
# omega_weight: 0.08
# omega_w_weight: 75

## dji last
# linear_weight: 8
# linear_w_weight: 2
# linear_R_weight: 0
# omega_weight: 0.08
# omega_w_weight: 75 / 80

## dji8-50 
# linear_weight: 1.2
# linear_w_weight: 1
# linear_R_weight: 0
# omega_weight: 0.06 # 1
# omega_w_weight: 12 # 15

output_file: "/home/hao/Desktop/radar-event-new/src/TwistEstimator/output/twist_estimator.csv" 
output_dt: 0.1
visualize_marker_scale: 0.8
visualize_doppler_scale: 1.5
cam_topic: "/dvs/image_raw"

local_dt: 3.0

# twist bias model
bias:
  omega: [0.0, 0.0, 0.0]
  vel: [0.0, 0.0, 0.0]
  # omega: [1.0, 2.0, 3.0]
  # vel: [1.0, 2.0, 3.0]